exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -252,6 +286,7 @@ private void setup() {
     }
 
     private void outputValues(IndexPointer outStartIdxHolder, IndexPointer outNumRecordsHolder) {
+      int batchOutputCount = 0;
```

Comment:
The original code had the intention to start at any offset in the batch. This change is hard-coding to always start at _zero_. If so, the first parameter (and the global **outStartIdxHolder**) can be eliminated.